### PR TITLE
Implement new rules valid from 5th of December 2016 & small refactoring

### PR DESCRIPTION
**Pull request by derschatta**
As we need to exchange our IBAN library with something new I added support for the newest rules which will be valid from December 2016 on.

I also changed the detection of the rule and revision to use the existing ruleset within this project and not rely on the Bav project anymore. I believe this is the better approach as you don't have to maintain two spereate projects to update the iban rules in the future.

### CODE REVIEW

**Comment by derschatta** [comment]
sry for the other closed pull requests, I realized it's better to make a pull request from my dev branch not from my master branch

File: library/IBAN/Generation/IBANGeneratorDE.php [hunk 1]
```diff
@@ -15,29 +14,38 @@
...
+
+        // bank not found
+        if (!isset($rules[$instituteIdentification])) {
+            return '';
+        }
+
+        $bank = $rules[$instituteIdentification];
+        $ibanRuleCodeAndVersion = $bank['rule'];
+
+        $ibanRule = $this->ruleFactory->createIbanRule($ibanRuleCodeAndVersion, $instituteIdentification, $bankAccountNumber);
```
Comment (jschaedl) [inline library/IBAN/Generation/IBANGeneratorDE.php]: 👍

File: tests/library/IBAN/Generation/IBANGeneratorTest.php [hunk 2]
```diff
@@ -311,6 +312,9 @@ public function testGenerateIbanForRuleDE002400()
 
     public function testGenerateIbanForRuleDE002500()
     {
+        if (IBANGeneratorDE::CURRENT_RULES >= 20161205) {
+            $this->markTestSkipped('No bank included in newest rules');
+        }
```
Comment (jschaedl) [inline tests/library/IBAN/Generation/IBANGeneratorTest.php]: I like the idea of skipping the tests. 👍

**Comment by jschaedl** [comment]
@derschatta Thanks a lot!

**Pull request merged**